controller/instance_manager_controller.go

```diff
@@ -2257,27 +2257,43 @@ func (m *InstanceManagerMonitor) isReplicaOrphaned(instanceName, node string) (b
 }
 
 // isInstanceOrphanedOnNode returns true only when it is very certain that an instance is scheduled on another node
-func (m *InstanceManagerMonitor) isInstanceOrphanedOnNode(spec *longhorn.InstanceSpec, status *longhorn.InstanceStatus, node string) bool {
-	if status.CurrentState != spec.DesireState {
+func (m *InstanceManagerMonitor) isInstanceOrphanedOnNode(procStatus *longhorn.InstanceProcessStatus, crMeta *metav1.ObjectMeta, crSpec *longhorn.InstanceSpec, crStatus *longhorn.InstanceStatus, node string) bool {
+	if !crMeta.DeletionTimestamp.IsZero() {
+		// Deletion in progress. Will handle this after finalized
 		return false
 	}
-	switch status.CurrentState {
+
+	if crStatus.CurrentState != crSpec.DesireState {
+		// status transitioning
+		return false
+	}
+
+	switch crStatus.CurrentState {
 	case longhorn.InstanceStateRunning:
-		return status.OwnerID == spec.NodeID && spec.NodeID != node
+		return crStatus.OwnerID == crSpec.NodeID && crSpec.NodeID != node
 	case longhorn.InstanceStateStopped:
-		return spec.NodeID != node
+		return crSpec.NodeID != node
 	default:
 		return false
 	}
 }
 
-func (m *InstanceManagerMonitor) createOrphanForInstances(existOrphans map[string]bool, im *longhorn.InstanceManager, instanceMap instanceProcessMap, orphanType longhorn.OrphanType, orphanFilter func(instanceName, node string) (bool, error)) {
+func (m *InstanceManagerMonitor) createOrphanForInstances(existOrphans map[string]bool, im *longhorn.InstanceManager, instanceMap instanceProcessMap, orphanType longhorn.OrphanType, orphanFilter func(process *longhorn.InstanceProcess, instanceName, node string) (bool, error)) {
 	for instanceName, instance := range instanceMap {
+		switch instance.Status.State {
+		case longhorn.InstanceStateStarting:
+			// Status transitioning. Will handle this after running.
+			continue
+		case longhorn.InstanceStateStopping, longhorn.InstanceStateStopped:
+			// Terminating. No orphan CR needed, and the orphaned instances will be cleanup by instance manager after stopped.
+			continue
```

Comment:
Should ignore `stopped`, isn't it?

Reply:
Yeah, so we list both `stopping` and `stopped` here.